### ISSUE TITLE
Cant login

### ISSUE THREAD

**Issue by karkaletsi** [UnamSanctam/UnamWebPanel#159]
basically i set up the web panel all good and when im trying to login it says "error something went wrong"
when i go in console and i see what the problem is its this:
![image](https://user-images.githubusercontent.com/104867512/176551332-1817fc2b-9800-4841-a6ed-8fb0da560a7d.png)

**Comment by UnamSanctam** [comment]
Make sure that you have PHP 7.4 or above, and also check for any error.log files anywhere and if there are any logfiles in `__UNAM_LIB\Logs`

**Comment by karkaletsi** [comment]
no log files. I have PHP 7.4.3 CLI

**Comment by UnamSanctam** [comment]
Make sure that you copied over all the files, you'll have to find out how to enable error logs for your specific host to see what causes the error 500.

**Comment by karkaletsi** [comment]
I will try updating PHP

**Comment by karkaletsi** [comment]
Can I make it so it auto logs me in?

**Comment by UnamSanctam** [comment]
Sure you can code it to do that but that wouldn't really be good, since then anyone that gets the URL won't have to log in either.

**Comment by karkaletsi** [comment]
Ok tell me what to change sir

**Comment by UnamSanctam** [comment]
Well you can change it to that but you'll need to fix the error first since there's something wrong in your configuration. Are you just creating a local web panel for testing or a public web panel which external computers will connect to?

**Comment by karkaletsi** [comment]
Public web panel which external computers will connect to.

**Comment by karkaletsi** [comment]
I will try to add an old version of your webpanel to see if it still does that

**Comment by karkaletsi** [comment]
still same error hmm

**Comment by UnamSanctam** [comment]
It's a PHP error, the jquery line is just from the AJAX response. You'll have to check your PHP configurations since it's most likely the cause, the easiest would be to enable errors.

**Comment by karkaletsi** [comment]
error_reporting(E_ALL);
ini_set('display_errors', '1');
put this in login.php and index.php 
and display_errors = On on both of my php.ini

**Comment by karkaletsi** [comment]
Where can i check for the errors, no file created on the logs folder

**Comment by UnamSanctam** [comment]
You should put it at the top of auth-ajax.php, then check all folders for an `error.log` file.

**Comment by karkaletsi** [comment]
Above the <?php right? I did that restarted apache tried logging in but no error.log anywhere

**Comment by UnamSanctam** [comment]
No below it.

**Comment by karkaletsi** [comment]
still nothing

**Comment by UnamSanctam** [comment]
It will be easier if you just set up something like WAMP (or XAMPP) instead, you only need Apache from WAMP and it will work.

**Comment by karkaletsi** [comment]
i guess i will leave the webpanel part, i remember the crypto miner startup didnt work did you manage to fix that?

**Comment by UnamSanctam** [comment]
I don't think it has ever not worked, except when you try to enter protected folders into "File Name" since it can't write to those.

**Comment by Quagliada3kg** [comment]
same problem here, can't login, no log, just the error in the photo.
I tried with apache+php and nginx+php. 
I'm running it on a rasperry pi 3b (just becasue it was there) on a lite (no screen) x64 OS (the defaoult one)

sorry for the bad english it's late for me and it isn't my primary language...

**Comment by UnamSanctam** [comment]
>same problem here, can't login, no log, just the error in the photo.
I tried with apache+php and nginx+php.
I'm running it on a rasperry pi 3b (just becasue it was there) on a lite (no screen) x64 OS (the defaoult one)
sorry for the bad english it's late for me and it isn't my primary language...

Make sure that you have PHP 8.0 or newer and that no files are missing.

**Comment by Quagliada3kg** [comment]
updated it to 8.2, still not working

**Comment by UnamSanctam** [comment]
>updated it to 8.2, still not working

Do you have any error logs inside the `__UNAM_LIB\Logs` folder or next to the `auth-ajax.php` file? Also, did you change `config.php` in any way?

**Comment by Quagliada3kg** [comment]
no error log anywhere, i changed only the password in config.php

**Comment by UnamSanctam** [comment]
>no error log anywhere, i changed only the password in config.php

Then possibly it doesn't have permission to write the log, if you replace the config.php with a fresh one from the releases https://github.com/UnamSanctam/UnamWebPanel/releases then does it work?

**Comment by Quagliada3kg** [comment]
nop

**Comment by UnamSanctam** [comment]
>nop

Then please post more information about the error and your setup, since your setup is the issue then it's hard to guess what the problem is with it.

**Comment by Quagliada3kg** [comment]
hardware:
- rasperry pi 3b connected with a net cable

software:
- Apache/2.4.56 (Debian)
- PHP 8.2.8 (cli) connected to Apache2 with libapache2-mod-php

i think it's all

**Comment by UnamSanctam** [comment]
Can you check phpinfo() and see what it says regarding PHP version and such? And did you try fully removing all the files, then -re-extracting all the web panel files and then testing logging in with the default password?

**Comment by Quagliada3kg** [comment]
![image](https://github.com/UnamSanctam/UnamWebPanel/assets/102754456/7f32eabd-c48e-4a90-b431-57e20b853e44)
![image](https://github.com/UnamSanctam/UnamWebPanel/assets/102754456/28bdc2b7-fc79-417f-ae5d-0031f212aaf8)
![image](https://github.com/UnamSanctam/UnamWebPanel/assets/102754456/99b18449-da76-47ee-bb0e-7f29e43af159)
![image](https://github.com/UnamSanctam/UnamWebPanel/assets/102754456/5f184af6-f504-4ebe-8d6b-46dbf1c2f7a7)

**Comment by UnamSanctam** [comment]
Do you have php-sqlite installed or not? And can you enable error outputting and then look at the response from auth-ajax.php?

**Comment by Quagliada3kg** [comment]
php-sqlite it's installed, how can i enable error outputting?

**Comment by UnamSanctam** [comment]
Try changing this to '`on`':
https://github.com/UnamSanctam/UnamWebPanel/blob/b8b30b7b0412fa2f817fb80df9b561291eddb186/UnamWebPanel/class/class.base.php#L4

**Comment by Quagliada3kg** [comment]
ok then?

**Comment by UnamSanctam** [comment]
Try to log in and see what auth-ajax.php responds with in the developer console (in the network tab).

**Comment by Quagliada3kg** [comment]
![image](https://github.com/UnamSanctam/UnamWebPanel/assets/102754456/e43d4d8f-0780-4ac3-a90c-e63d2519aa79)

**Comment by UnamSanctam** [comment]
Seems like you haven't extracted all the web panel files.

**Comment by Quagliada3kg** [comment]
code 200 all ok
but other error here

![image](https://github.com/UnamSanctam/UnamWebPanel/assets/102754456/ef7b3fc8-b634-40b2-8b95-a3fabe3b89be)

**Comment by UnamSanctam** [comment]
What does auth-ajax.php return? The CSS file doesn't matter since it's only a stylesheet.

**Comment by Quagliada3kg** [comment]
`<br />
<b>Warning</b>:  file_put_contents(/var/www/html/__UNAM_LIB/Logs/php-error-08-04-2023.html): Failed to open stream: Permission denied in <b>/var/www/html/__UNAM_LIB/unam_lib.php</b> on line <b>235</b><br />
<br />
<b>Fatal error</b>:  Uncaught Error: Call to a member function prepare() on bool in /var/www/html/__UNAM_LIB/unam_lib.php:29
Stack trace:
#0 /var/www/html/auth-ajax.php(11): unam_lib-&gt;unam_dbSelect()
#1 {main}
  thrown in <b>/var/www/html/__UNAM_LIB/unam_lib.php</b> on line <b>29</b><br />`

**Comment by UnamSanctam** [comment]
Yes then it wasn't allowed to write the errors like I thought (which is why there's no log), seems like the problem is that you're not able to open the database. Make sure that the PDO SQLite extension is both installed and enabled and that the `db/unamwebpanel.db` file and parent folder are both readable and writable.

**Comment by Quagliada3kg** [comment]
db/unamwebpanel.db have the permission, and SQLite is up and running

**Comment by UnamSanctam** [comment]
Give write permissions to the `__UNAM_LIB/Logs` folder so that it can write out any errors into that folder. But something is causing the database not to work, since I don't have access to the machine then it's hard to know why. Can you see the SQLite extension active inside phpinfo()?

**Comment by Quagliada3kg** [comment]
sorry if i disappeared but i had health problems, the folder has all permissions (they were pre-set) the sqlite info i can find on phpinfo() is this:

![image](https://github.com/UnamSanctam/UnamWebPanel/assets/102754456/09d20e47-678b-46ad-a4cb-33ded038b32e)
![image](https://github.com/UnamSanctam/UnamWebPanel/assets/102754456/e872cedf-33c0-4642-9041-a7b4c1f5ca63)

**Comment by Quagliada3kg** [comment]
tried on a different pi3, fresh install and all, this times a program comes out on auth-ajax.php

![image](https://github.com/UnamSanctam/UnamWebPanel/assets/102754456/278f2310-29c0-4ae5-890d-12fd710c462a)

**Comment by UnamSanctam** [comment]
Your server is likely unable to start/connect the SQLite database. Can you set write permissions for both the `class` and `__UNAM_LIB/Logs` folders? Then try to login and check if there's an error log in either the `class` folder or the `__UNAM_LIB/Logs` folder.

**Comment by Quagliada3kg** [comment]
this is the situation with permissions at the moment, no log files or anything

![image](https://github.com/UnamSanctam/UnamWebPanel/assets/102754456/541d93be-f88a-49d1-9fc4-58272c2f1c78)

**Comment by UnamSanctam** [comment]
Set everything to 777 and see if something happens, that's usually how you debug. Then for production you can set them to only the required ones.

**Comment by Quagliada3kg** [comment]
and it works!

**Comment by UnamSanctam** [comment]
Alright great, then it's due to the permissions. It really depends on how the Apache and Debian is set up, the database file and folder should typically be owned by the group that Apache is running under. Usually this is the `www-data` user on Debian. If the user is www-data and the folder you're hosting the web panel inside is `/var/www` then you could try these commands (I just wrote them from memory so hopefully they're alright):

```
cd /var/www
chown -R $USER:www-data .
find . -type d -exec chmod 755 {} \;
find . -type f -exec chmod 644 {} \;
chown -R www-data:www-data /var/www/db
chmod -R 700 /var/www/db
```

**Comment by Quagliada3kg** [comment]
error 403 forbidden

**Comment by UnamSanctam** [comment]
Is the web panel directly inside `/var/www`? What are all the permissions and ownership set to currently? And do you know if the Apache server is running under www-data or some other user?

**Comment by Quagliada3kg** [comment]
fixed, the site was in a subfolder under `/www`

**Comment by UnamSanctam** [comment]
Alright great, is it all working now?
